crates/codegen/runtime/cargo/wasm/src/runtime/wrappers/bindings/mod.rs

```diff
@@ -0,0 +1,174 @@
+use crate::wasm_crate::utils::{define_rc_wrapper, define_wrapper, IntoFFI};
+
+mod ffi {
+    pub use crate::wasm_crate::bindgen::exports::nomic_foundation::slang::bindings::{
+        BindingGraph, BindingGraphBorrow, BindingLocation, BuiltInLocation, BuiltInLocationBorrow,
+        CursorBorrow, Definition, DefinitionBorrow, Guest, GuestBindingGraph, GuestBuiltInLocation,
+        GuestDefinition, GuestReference, GuestUserFileLocation, Reference, ReferenceBorrow,
+        UserFileLocation, UserFileLocationBorrow,
+    };
+    pub use crate::wasm_crate::bindgen::exports::nomic_foundation::slang::cst::Cursor;
+}
+
+mod rust {
+    pub use crate::rust_crate::bindings::{
+        BindingGraph, BindingLocation, BuiltInLocation, UserFileLocation,
+    };
+
+    /// TODO: This is a work-around for the fact that `metaslang_bindings` internals (handles, locators, etc...) are exposed.
```

Comment:
@ggiraldez these are the wrappers I mentioned, due to the fact that the original `Definition` and `Reference` types have handles in them.
They should be removed once the `metaslang_bindings` API is fixed.

Reply:
Is the problem we don't want to expose those handles to Typescript? The handles themselves are `usize` values. Or is the pointer/reference to the bindings object that's the problem? I recall you mentioning that they should be copyable.

Since we're going to pre-resolve all references anyway, we may just get away with cursors in the end. But asking to understand and to be aware of what are the limitations.

Reply:
> is the pointer/reference to the bindings object that's the problem? 

Yes. For WIT resources, we need to track their lifetimes on the other side of the FFI boundary, and there isn't a straightforward way to do that with the handle references.
For the existing WIT resources, we either wrap them in `Rc<>` (`define_rc_wrapper()`) or `RefCell<>` (`define_refcell_wrapper()`).